src/rpc/ngHandlers/NFTBuyOffers.cpp

```diff
@@ -0,0 +1,226 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <rpc/RPCHelpers.h>
+#include <rpc/ngHandlers/NFTBuyOffers.h>
+
+#include <ripple/app/tx/impl/details/NFTokenUtils.h>
+#include <ripple/protocol/Indexes.h>
+
+using namespace ripple;
+using namespace ::RPC;
+
+namespace ripple {
+
+// TODO: move to some common serialization impl place
+inline void
+tag_invoke(
+    boost::json::value_from_tag,
+    boost::json::value& jv,
+    SLE const& offer)
+{
+    auto amount = ::RPC::toBoostJson(
+        offer.getFieldAmount(sfAmount).getJson(JsonOptions::none));
+
+    boost::json::object obj = {
+        {JS(nft_offer_index), to_string(offer.key())},
+        {JS(flags), offer[sfFlags]},
+        {JS(owner), toBase58(offer.getAccountID(sfOwner))},
+        {JS(amount), std::move(amount)},
+    };
+
+    if (offer.isFieldPresent(sfDestination))
+        obj.insert_or_assign(
```

Comment:
other places use [], wonder why we use insert_or_assign here?

Reply:
I think it's more efficient than [] either way and we probably should use it more. But as with other things - this code is just copied as is.